dkg/store.go

```diff
@@ -14,8 +18,9 @@ import (
 
 type boltStore struct {
 	sync.RWMutex
-	db  *bolt.DB
-	log log.Logger
+	db            *bolt.DB
+	log           log.Logger
+	migrationLock sync.Mutex
```

Comment:
Do we really need a `migrationLock`? 😢

Reply:
Eughh yeah unfortunately - access to the `map[string]Broadcast` needs to be protected unfort.
I think when we move to gossip mode we might need to even lock to process each packet so we don't do:
```
current = getCurrentState()
finalState = applyChange(current)
store(finalState)
```

Although I haven't yet looked into bolt transactions for that scenario, they might be a better choice